### PR TITLE
Accounts umask

### PR DESCRIPTION
**Pull request by jonathan-sturges**
#### Description:

- Add Ignition support for setting user umask

#### Rationale:

- _Rationale here. Replace this text. Don't use the italics format!_

- Fixes # _Issue number here (e.g. #26) or remove this line if no issue exists._

### CODE REVIEW

**Comment by shawndwells** [comment]
@openscap-ci add to whitelist

**Comment by jonathan-sturges** [comment]
Shawn, I know I don't have CCEs referenced.  Can you help me with that, and let me know if you see other missing pieces.  Thanks!

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        filesystem: root
+        mode: 0644
+        path: /etc/bashrc
```
Comment (yuumasato) [inline linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml:18]: Please, make sure the files end  with a newline character. Note the red marks in the files.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_csh_cshrc/ignition/shared.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
...%3A%25c%5C033%5C%5C%25%7D%5B%25n%40%25m%20%25c%5D%25%23%20%27%0A%20%20%20%20%20%20%20%20endif%0A%20%20%20%20%20%20%20%20breaksw%0A%20%20%20%20%20%20default%3A%0A%20%20%20%20%20%20%20%20breaksw%0A%20%20%20%20endsw%0A%20%20endif%0Aendif%0A%0Asetenv%20MAIL%20%22/var/spool/mail/%24USER%22%0A%0A%23%20Check%20if%20we%20aren%27t%20a%20loginshell%20and%20do%20stuff%20if%20we%20aren%27t%0Aif%20%28%21%20%24%3Floginsh%29%20then%0A%20%20%20%20if%20%28%20-d%20/etc/profile.d%20%29%20then%0A%20%20%20%20%20%20%20%20set%20nonomatch%0A%20%20%20%20%20%20%20%20foreach%20i%20%28%20/etc/profile.d/%2A.csh%20%29%0A%20%20%20%20%20%20%20%20%20%20%20%20if%20%28%20-r%20%22%24i%22%20%29%20then%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20if%20%28%24%3Fprompt%29%20then%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20source%20%22%24i%22%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20else%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20source%20%22%24i%22%20%3E%26/dev/null%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20endif%0A%20%20%20%20%20%20%20%20%20%20%20%20endif%0A%20%20%20%20%20%20%20%20end%0A%20%20%20%20%20%20%20%20unset%20i%20nonomatch%0A%20%20%20%20endif%0Aendif%0A
```
Comment (yuumasato) [inline linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_csh_cshrc/ignition/shared.yml:15]: When the content is built `{{{ var_accounts_user_umask }}}` is rendered empty, like this: `%20umask%20%0Aelse`.

I think the correct value is:
`<xccdf-1.2:sub idref="xccdf_org.ssgproject.content_value_var_accounts_user_umask" use="legacy"/>`
which will be substituted by the scanner when rendering out the fix snippet.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml (hunk 1 above)
Comment (jonathan-sturges) [reply linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml]: done, thanks

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_csh_cshrc/ignition/shared.yml (hunk 2 above)
Comment (jonathan-sturges) [reply linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_csh_cshrc/ignition/shared.yml]: changed, thanks

**Comment by jonathan-sturges** [comment]
Ahh, I misunderstood the order of operations wrt decoding files & substituting vars.  Let me fix this up real quick...

**Comment by jonathan-sturges** [comment]
> @jonathan-sturges: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> ci/prow/e2e-aws-e8	[b3c47ac](https://github.com/ComplianceAsCode/content/commit/b3c47acdaed27097569ea3f8d0cefe9f86c9231e)	[link](https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/ComplianceAsCode_content/5630/pull-ci-ComplianceAsCode-content-master-e2e-aws-e8/54)	`/test e2e-aws-e8`
> [Full PR test history](https://prow.svc.ci.openshift.org/pr-history?org=ComplianceAsCode&repo=content&pr=5630). [Your PR dashboard](https://prow.svc.ci.openshift.org/pr?query=is:pr+state:open+author:jonathan-sturges). Please help us cut down on flakes by [linking to](https://github.com/kubernetes/community/blob/master/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/ComplianceAsCode/content/issues?q=is:issue+is:open) when you hit one in your PR.

/retest

**Comment by matejak** [comment]
I see that @yuumasato review has been probably addressed, and @shawndwells reviews something that is not part of the PR any more.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
...are/doc/setup-%2A/uidgid%20file%0A%20%20%20%20if%20%5B%20%24UID%20-gt%20199%20%5D%20%26%26%20%5B%20%22%60id%20-gn%60%22%20%3D%20%22%60id%20-un%60%22%20%5D%3B%20then%0A%20%20%20%20%20%20%20umask%20%7B%7B%7B%20var_accounts_user_umask%20%7D%7D%7D%0A%20%20%20%20else%0A%20%20%20%20%20%20%20umask%20%7B%7B%7B%20var_accounts_user_umask%20%7D%7D%7D%0A%20%20%20%20fi%0A%0A%20%20%20%20SHELL%3D/bin/bash%0A%20%20%20%20%23%20Only%20display%20echos%20from%20profile.d%20scripts%20if%20we%20are%20no%20login%20shell%0A%20%20%20%20%23%20and%20interactive%20-%20otherwise%20just%20process%20them%20to%20set%20envvars%0A%20%20%20%20for%20i%20in%20/etc/profile.d/%2A.sh%3B%20do%0A%20%20%20%20%20%20%20%20if%20%5B%20-r%20%22%24i%22%20%5D%3B%20then%0A%20%20%20%20%20%20%20%20%20%20%20%20if%20%5B%20%22%24PS1%22%20%5D%3B%20then%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20.%20%22%24i%22%0A%20%20%20%20%20%20%20%20%20%20%20%20else%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20.%20%22%24i%22%20%3E/dev/null%0A%20%20%20%20%20%20%20%20%20%20%20%20fi%0A%20%20%20%20%20%20%20%20fi%0A%20%20%20%20done%0A%0A%20%20%20%20unset%20i%0A%20%20%20%20unset%20-f%20pathmunge%0A%20%20fi%0A%0Afi%0A%23%20vim%3Ats%3D4%3Asw%3D4%0A
```
Comment (JAORMX) [inline linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml:11]: Currently variable substitution doesn't work for the compliance-operator (rhcos4 and ocp4 content)... So, you would need to choose a hardcoded value to make this work currently. Else the ` {{{ var_accounts_user_umask }}}` variable you're trying to use won't work and will just get literaly written to disk in the RHCOS host, which won't work either.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/rule.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: ol7,ol8,rhel6,rhel7,rhel8,sle15
+prodtype: ocp4,ol7,ol8,rhel6,rhel7,rhel8,sle15
```
Comment (JAORMX) [inline linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/rule.yml:3]: we're using rhcos4 now as the product for this.

**Comment by jonathan-sturges** [comment]
ACK-ing this.  Give me a day or 2 to complete.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml (hunk 3 above)
Comment (jonathan-sturges) [reply linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/ignition/shared.yml]: Thank you, I've changed it to a fixed value of '027', which is the default for var_accounts_user_umask.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/rule.yml (hunk 4 above)
Comment (jonathan-sturges) [reply linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_bashrc/rule.yml]: ACK'd and done.

**Comment by matejak** [comment]
Merging, as all reviewers comments were addressed. Thanks!

**Pull request merged**